lib/scraped_page/request.rb

```diff
@@ -0,0 +1,28 @@
+require 'scraped_page/strategy/live_request'
+require 'scraped_page/response'
+
+class ScrapedPage
+  class Request
+    def initialize(url:, readers: [Strategy::LiveRequest.new], writers: [])
+      @url = url
+      @readers = readers
+      @writers = writers
+    end
+
+    def response
+      abort "Couldn't find a response for #{url}" if readers_response.nil?
+      response = Response.new(readers_response.merge(url: url))
+      writers.each { |w| w.save_response(response) }
+      response
+    end
+
+    private
+
+    def readers_response(url)
```

Comment:
perhaps something like `first_successful_response` might be a bit more explicit?